src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,172 @@
+"""ABY3 Protocol.
+
+ABY3 : A Mixed Protocol Framework for Machine Learning.
+https://eprint.iacr.org/2018/403.pdf
+"""
+# stdlib
+from typing import Any
+from typing import List
+from typing import Tuple
+
+# third party
+import torch
+import torchcsprng as csprng
+
+from sympc.protocol.protocol import Protocol
+from sympc.session import Session
+from sympc.tensor import MPCTensor
+from sympc.tensor import ReplicatedSharedTensor
+from sympc.tensor.tensor import SyMPCTensor
+
+gen = csprng.create_random_device_generator()
+
+
+class ABY3(metaclass=Protocol):
+    """ABY3 Protocol Implementation."""
+
+    """Used for Share Level static operations like distributing the shares."""
+    share_class: SyMPCTensor = ReplicatedSharedTensor
+    security_levels: List[str] = ["semi-honest", "malicious"]
+
+    def __init__(self, security_type: str = "semi-honest"):
+        """Initialization of the Protocol.
+
+        Args:
+            security_type : specifies the security level of the Protocol.
+
+        Raises:
+            ValueError : If invalid security_type is provided.
+        """
+        if security_type not in self.security_levels:
+            raise ValueError(f"{security_type} is not a valid security type")
+
+        self.security_type = security_type
+
+    def __eq__(self, other: Any) -> bool:
+        """Check if "self" is equal with another object given a set of attributes to compare.
+
+        Args:
+            other (Any): Object to compare
+
+        Returns:
+            bool: True if equal False if not.
+        """
+        if self.security_type != other.security_type:
+            return False
+
+        if type(self) != type(other):
+            return False
+
+        return True
+
+    @staticmethod
+    def truncation_algorithm1(
```

Comment:
Q: Can't we use parallel execution for this? (and in that case, we won't have to locally get the shares)

Reply:
I wanted to do parallel execution at start, in that scenario parties mutually communicate with each other ,we require a P2P for that,share truncation algorithm 1 involves increased round complexity ,when we execute it without, I highly optimized it for our orchestrator model, so we could have only one round complexity. Eventually when we move to truncation `2` ,we delegate it locally using parallel execution.